addons/mail/static/src/model/model_field.js

```diff
@@ -276,13 +276,11 @@ class ModelField {
             if (['one2one', 'many2one'].includes(this.relationType)) {
                 return OtherModel.get(this.read(record));
             }
-            return this.read(record)
-                .map(localId => OtherModel.get(localId))
-                /**
-                 * FIXME: Stored relation may still contain
-                 * outdated records.
-                 */
-                .filter(record => !!record);
+            const res = [];
+            for (const localId of this.read(record)) {
+                res.push(OtherModel.get(localId));
+            }
+            return res;
```

Comment:
no risk to have `undefined` items in array (due to outdated records)?

Reply:
Do you see any particular way to get into this situation?

Delete is supposed to unlink relational fields, and link methods are supposed to filter out deleted records.

Reply:
During a setRelation, could have specific moment in which value contains outdated records. Could not reproduce with exact example, some tests just fail. I think tests that had is causal were candidates. Runbot's green so should be fine.